src/toolchain.rs

```diff
@@ -972,6 +908,68 @@ impl<'a> DistributableToolchain<'a> {
     }
 }
 
+/// Helper type to avoid parsing a manifest more than once
+pub(crate) struct ToolchainDescWithManifest {
+    toolchain: ToolchainDesc,
+    manifestation: Manifestation,
+    pub(crate) manifest: Manifest,
```

Comment:
With the struct being `pub(crate)` there's no need for the `(crate)` modifier on this field.